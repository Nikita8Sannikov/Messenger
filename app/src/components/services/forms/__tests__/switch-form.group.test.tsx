import { fireEvent, render, screen } from "@testing-library/react";

import SwitchFormGroup from "@components/services/forms/switch-form-group";

describe("SwitchFormGroup", () => {
	const mockForm = {
		soundEnabled: true,
		messageSound: false,
		friendRequestSound: true,
	};

	const mockOnChange = jest.fn();

	beforeEach(() => {
		jest.clearAllMocks();
	});

	it("matches snapshot with default props", () => {
		const { container } = render(<SwitchFormGroup form={mockForm} onChange={mockOnChange} />);
		expect(container).toMatchSnapshot();
	});

	it("matches snapshot with all switches enabled", () => {
		const allEnabledForm = {
			soundEnabled: true,
			messageSound: true,
			friendRequestSound: true,
		};
		const { container } = render(<SwitchFormGroup form={allEnabledForm} onChange={mockOnChange} />);
		expect(container).toMatchSnapshot();
	});

	it("matches snapshot with all switches disabled", () => {
		const allDisabledForm = {
			soundEnabled: false,
			messageSound: false,
			friendRequestSound: false,
		};
		const { container } = render(<SwitchFormGroup form={allDisabledForm} onChange={mockOnChange} />);
		expect(container).toMatchSnapshot();
	});

	it("renders all switches with correct labels", () => {
		render(<SwitchFormGroup form={mockForm} onChange={mockOnChange} />);

		expect(screen.getByLabelText("All sound notifications")).toBeInTheDocument();
		expect(screen.getByLabelText("Sound notifications when receiving a message")).toBeInTheDocument();
		expect(screen.getByLabelText("Sound notifications when receiving a friend's request")).toBeInTheDocument();
	});

	it("renders switches with correct initial states", () => {
		render(<SwitchFormGroup form={mockForm} onChange={mockOnChange} />);

		const soundEnabledSwitch = screen.getByLabelText("All sound notifications");
		const messageSoundSwitch = screen.getByLabelText("Sound notifications when receiving a message");
		const friendRequestSoundSwitch = screen.getByLabelText("Sound notifications when receiving a friend's request");

		expect(soundEnabledSwitch).toBeChecked();
		expect(messageSoundSwitch).not.toBeChecked();
		expect(friendRequestSoundSwitch).toBeChecked();
	});

	it("calls onChange with correct parameters when soundEnabled is toggled", () => {
		render(<SwitchFormGroup form={mockForm} onChange={mockOnChange} />);

		const soundEnabledSwitch = screen.getByLabelText("All sound notifications");
		fireEvent.click(soundEnabledSwitch);

		expect(mockOnChange).toHaveBeenCalledWith("soundEnabled", false);
	});

	it("calls onChange with correct parameters when messageSound is toggled", () => {
		render(<SwitchFormGroup form={mockForm} onChange={mockOnChange} />);

		const messageSoundSwitch = screen.getByLabelText("Sound notifications when receiving a message");
		fireEvent.click(messageSoundSwitch);

		expect(mockOnChange).toHaveBeenCalledWith("messageSound", true);
	});

	it("calls onChange with correct parameters when friendRequestSound is toggled", () => {
		render(<SwitchFormGroup form={mockForm} onChange={mockOnChange} />);

		const friendRequestSoundSwitch = screen.getByLabelText("Sound notifications when receiving a friend's request");
		fireEvent.click(friendRequestSoundSwitch);

		expect(mockOnChange).toHaveBeenCalledWith("friendRequestSound", false);
	});

	it("updates switch states when form prop changes", () => {
		const { rerender } = render(<SwitchFormGroup form={mockForm} onChange={mockOnChange} />);

		const newForm = {
			soundEnabled: false,
			messageSound: true,
			friendRequestSound: false,
		};

		rerender(<SwitchFormGroup form={newForm} onChange={mockOnChange} />);

		const soundEnabledSwitch = screen.getByLabelText("All sound notifications");
		const messageSoundSwitch = screen.getByLabelText("Sound notifications when receiving a message");
		const friendRequestSoundSwitch = screen.getByLabelText("Sound notifications when receiving a friend's request");

		expect(soundEnabledSwitch).not.toBeChecked();
		expect(messageSoundSwitch).toBeChecked();
		expect(friendRequestSoundSwitch).not.toBeChecked();
	});
});
