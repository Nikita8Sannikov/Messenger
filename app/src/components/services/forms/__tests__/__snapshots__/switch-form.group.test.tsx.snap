// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchFormGroup matches snapshot with all switches disabled 1`] = `
<div>
  <div
    class="MuiFormGroup-root settings-form css-1mac7pt-MuiFormGroup-root"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            name="soundEnabled"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
      >
        All sound notifications
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            name="messageSound"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
      >
        Sound notifications when receiving a message
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            name="friendRequestSound"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
      >
        Sound notifications when receiving a friend's request
      </span>
    </label>
  </div>
</div>
`;

exports[`SwitchFormGroup matches snapshot with all switches enabled 1`] = `
<div>
  <div
    class="MuiFormGroup-root settings-form css-1mac7pt-MuiFormGroup-root"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            name="soundEnabled"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
      >
        All sound notifications
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            name="messageSound"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
      >
        Sound notifications when receiving a message
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            name="friendRequestSound"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
      >
        Sound notifications when receiving a friend's request
      </span>
    </label>
  </div>
</div>
`;

exports[`SwitchFormGroup matches snapshot with default props 1`] = `
<div>
  <div
    class="MuiFormGroup-root settings-form css-1mac7pt-MuiFormGroup-root"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            name="soundEnabled"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
      >
        All sound notifications
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            name="messageSound"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
      >
        Sound notifications when receiving a message
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            name="friendRequestSound"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
      >
        Sound notifications when receiving a friend's request
      </span>
    </label>
  </div>
</div>
`;
